### PR TITLE
[REVIEW] Move GPU [de]compressors to cudf::io namespace

### PR DESCRIPTION
**Pull request by OlivierNV**
Move compressors & decompressors to cudf::io namespace, share commonly used block-level macros & helpers.

### CODE REVIEW

File: cpp/src/io/utilities/block_utils.cuh [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+ * limitations under the License.
+ */
+
+#pragma once
+#include <stdint.h>
+
+namespace cudf {
+namespace io {
+
+#if (__CUDACC_VER_MAJOR__ >= 9)
```
Comment (jrhemstad) [inline cpp/src/io/utilities/block_utils.cuh]: Since our minimum CUDA version is 9.2 anyways, why not get rid of the macros all together?

Comment (OlivierNV) [reply cpp/src/io/utilities/block_utils.cuh]: I actually still do some development on Windows with cuda8 on Maxwell, where I use the same source files :)

**Comment by OlivierNV** [comment]
rerun tests

File: cpp/src/io/utilities/block_utils.cuh (hunk 1 above)
Comment (mjsamoht) [reply cpp/src/io/utilities/block_utils.cuh]: Okay, let's keep this then until you move your Windows environment to CUDA 9.2 or later.

**Pull request merged**